tests/jballerina-unit-test/src/test/resources/test-src/object/object-with-defaultable-field.bal

```diff
@@ -18,3 +18,43 @@ class Person {
         self.month = val1;
     }
 }
+
+final int classI = 111222;
+
+class ModuleVariableReferencingClass {
+    int i = classI;
+}
+
+function value(int k = classI) returns int {
+    return k;
+}
+
+ModuleVariableReferencingClass c1 = new;
+
+function testClassWithModuleLevelVarAsDefaultValue() {
+    ModuleVariableReferencingClass c = new;
+    assertEquality(111222, c.i);
+    assertEquality(111222, c1.i);
+}
+
+function testObjectWithModuleLevelVarAsDefaultValue() {
+    var value = object {
+        int i = classI;
+    };
+    assertEquality(111222, value.i);
+}
+
+const ASSERTION_ERROR_REASON = "AssertionError";
+
+function assertEquality(anydata expected, anydata actual) {
+    if expected is anydata && actual is anydata && expected == actual {
```

Comment:
Don't you get warnings for the checks now? Always true?

Reply:
Please update the logic, removing unnecessary checks.

Reply:
didn't get the warning.

Reply:
Hint rather. Didn't you get `unnecessary condition: expression will always evaluate to 'true'`?